src/test/java/org/mastodon/mamut/feature/dimensionalityreduction/umap/util/StandardScalerTest.java

```diff
@@ -0,0 +1,85 @@
+package org.mastodon.mamut.feature.dimensionalityreduction.umap.util;
+
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.assertArrayEquals;
+
+class StandardScalerTest
+{
+	@Test
+	void standardizeColumns_withMultipleColumns()
+	{
+		double[][] array = {
+				{ 1, 1, 1 },
+				{ 2, 2, 2 },
+				{ 3, 3, 3 },
+				{ 4, 4, 4 },
+				{ 5, 5, 5 }
+		};
+		double[][] expected = {
+				{ -1.2649110640673518, -1.2649110640673518, -1.2649110640673518 },
+				{ -0.6324555320336759, -0.6324555320336759, -0.6324555320336759 },
+				{ 0, 0, 0 },
+				{ 0.6324555320336759, 0.6324555320336759, 0.6324555320336759 },
+				{ 1.2649110640673518, 1.2649110640673518, 1.2649110640673518 }
+		};
+		StandardScaler.standardizeColumns( array );
+		assertArrayEquals( expected, array );
+	}
+
+	@Test
+	void standardizeColumns_withSingleColumn()
+	{
+		double[][] array = {
+				{ 1 },
+				{ 2 },
+				{ 3 },
+				{ 4 },
+				{ 5 }
+		};
+		double[][] expected = {
+				{ -1.2649110640673518 },
+				{ -0.6324555320336759 },
+				{ 0 },
+				{ 0.6324555320336759 },
+				{ 1.2649110640673518 }
+		};
+		StandardScaler.standardizeColumns( array );
+		assertArrayEquals( expected, array );
+	}
+
+	@Test
+	void standardizeColumns_withSingleRow()
+	{
+		double[][] array = { { 1, 2, 3 } };
+		double[][] expected = { { 0, 0, 0 } };
+		StandardScaler.standardizeColumns( array );
+		assertArrayEquals( expected, array );
+	}
+
+	@Test
+	void standardizeColumns_withEmptyArray()
+	{
+		double[][] array = {};
+		double[][] expected = {};
+		StandardScaler.standardizeColumns( array );
+		assertArrayEquals( expected, array );
+	}
+
+	@Test
+	void standardizeColumns_withZeroVarianceColumn()
+	{
+		double[][] array = {
+				{ 3, 3, 3 },
+				{ 3, 3, 3 },
+				{ 3, 3, 3 }
+		};
+		double[][] expected = {
+				{ 0, 0, 0 },
+				{ 0, 0, 0 },
+				{ 0, 0, 0 }
+		};
+		StandardScaler.standardizeColumns( array );
+		assertArrayEquals( expected, array );
```

Comment:
I would prefer if only one of the three columns has variance zero. To make sure the other columns are still treated correctly.